Stack/Opc.Ua.Core/Types/Encoders/JsonEncoder.cs

```diff
@@ -1320,14 +1321,27 @@ public void WriteExtensionObject(string fieldName, ExtensionObject value)
 
             PushStructure(fieldName);
 
+            var typeId = value.TypeId;
+                     
+            if (encodeable != null)
+            {
+                switch (value.Encoding)
+                {
+                    case ExtensionObjectEncoding.Binary: { typeId = encodeable.BinaryEncodingId; break; }
+                    case ExtensionObjectEncoding.Xml: { typeId = encodeable.XmlEncodingId; break; }
+                    default: { typeId = encodeable.TypeId; break; }
+                }
+            }
+
+            var localTypeId = ExpandedNodeId.ToNodeId(typeId, Context.NamespaceUris);
+
```

Comment:
nitpick: localType is only used for the reversible case and can be moved in the if case